src/random.cc

```diff
@@ -3,7 +3,6 @@
 
 #include "containers/scoped.hpp"
```

Comment:
Having includes in a specific order is mostly superstitious, beyond just being organized, but if we're going to put `<memory>` back, this is where it would go (in this project).  As a system include it's at the top of the list, except for the cpp's own header file.

```suggestion
#include <memory>

#include "containers/scoped.hpp"
```